src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts

```diff
@@ -0,0 +1,103 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from "vscode-azureextensionui";
+import { getExtensionPath, longRunning }  from '../utils/host';
+import { AppLensARMResponse, getAppLensDetectorData } from '../utils/detectors';
+import * as fs from 'fs';
+import * as htmlhandlers from "handlebars";
+import path = require('path');
+import { htmlHandlerRegisterHelper } from '../utils/detectorhtmlhelpers';
+import AksClusterTreeItem from '../../tree/aksClusterTreeItem';
+
+export default async function aksCRUDDiagnostics(
+    context: IActionContext,
+    target: any
+): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    if (cloudExplorer.available) {
+      const cloudTarget = cloudExplorer.api.resolveCommandTarget(target);
+
+      if (cloudTarget && cloudTarget.cloudName === "Azure" &&
+            cloudTarget.nodeType === "resource" && cloudTarget.cloudResource.nodeType === "cluster") {
+              const cluster = cloudTarget.cloudResource as AksClusterTreeItem;
+              const extensionPath = getExtensionPath();
+              if (extensionPath && cluster) {
+                await loadDetector(cluster, extensionPath);
+              }
+        } else {
+          vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        }
+    }
+}
+
+async function loadDetector(
+    cloudTarget: AksClusterTreeItem,
+    extensionPath: string) {
+    const clustername = cloudTarget.name;
+
+    await longRunning(`Loading ${clustername} diagnostics.`,
+          async () => {
+            const clusterAppLensData = await getAppLensDetectorData(cloudTarget, "aks-category-crud");
+            const detectorMap = new Map();
```

Comment:
See suggestions in https://github.com/Tatsinnit/vscode-aks-tools/pull/3/files

Not major, and could be a separate PR, but given this is a new file it'd maybe make sense to keep it as part of this one, if they make sense!

Reply:
💡☕️ cool, let’s merge this any other refactor we can do that as separate refactor so that feature PR stays more focused towards aks diagnostic crud feature. 😊